application/controllers/UserManagementController.php

```diff
@@ -420,6 +420,204 @@ public function actionDeleteUser()
+        $results = $this->unlimitedUserActivation($userIds, $status);
+
+        $error = null;
+        foreach ($results as $result) {
+            if (isset($result['error']) && $result['error'] == 'Reached the limit') {
+                $error = true;
+            }
+        }
+
+        $tableLabels = array(gT('User ID'), gT('Username'), gT('Status'));
+
+        $data = array(
+            'aResults'     => $results,
+            'successLabel' => gT('Saved successfully'),
+            'tableLabels' =>  $tableLabels,
+        );
+
+        if ($error) {
+            $data['additionalMessage'] = $this->renderPartial('/userManagement/partial/planupgradebutton');
+        }
+
+        Yii::app()->getController()->renderPartial('ext.admin.survey.ListSurveysWidget.views.massive_actions._action_results', $data);
+    }
+
+    /**
+     * Get the users limit & active users count
+     *
+     * @return string | JSON
+     * @throws CException
+     */
+    public function actionUserLimit()
+    {
+        if (!Permission::model()->hasGlobalPermission('users', 'update')) {
+            return $this->renderPartial(
+                'partial/error',
+                ['errors' => [gT("You do not have permission to access this page.")], 'noButton' => true]
+            );
+        }
+
+        $oEvent = new PluginEvent('beforeAdminUserActivation');
+        $oEvent->set('request', App()->request);
+        App()->getPluginManager()->dispatchEvent($oEvent);
+        $limit = $oEvent->get('limit', INF);
+        $count = $oEvent->get('count', 0);
+
+        $data = [
+            'active_user' => $count,
+            'user_limit' => $limit
+        ];
+        return App()->getController()->renderPartial('/admin/super/_renderJson', [
+            "data" => $data
+        ]);
+    }
+
+
+    /**
+     * Activate / deactivate user
+     *
+     * @return array
+     */
+    public function unlimitedUserActivation($userIds, $operation)
+    {
+        $results = [];
+        foreach ($userIds as $iUserId) {
+            $oUser = User::model()->findByPk($iUserId);
+            $results[$iUserId]['title'] = $oUser->users_name;
+            if (!$this->isAllowedToEdit($oUser)) {
+                $results[$iUserId]['error'] = gT('Unauthorized');
+                $results[$iUserId]['result'] = false;
+                continue;
+            }
+            $results[$iUserId]['result'] = $oUser->setActivationStatus($operation);
+            if (!$results[$iUserId]['result']) {
+                $results[$iUserId]['error'] = gT('Error');
+            }
+        }
+        return $results;
+    }
+
+    /**
+     * Handle user activation with regard to users limit
+     *
+     * @return array
+     */
+    public function limitedUserActivation($userIds, $operation, $limit)
+    {
+        $results = [];
+        foreach ($userIds as $iUserId) {
+            $oUser = User::model()->findByPk($iUserId);
+            $results[$iUserId]['title'] = $oUser->users_name;
```

Comment:
Same here, check for null :)

Reply:
Or wait, maybe this function should be moved to cloud plugin instead? Because core LS does not care about limits.

Reply:
Which also means the "unlimited" above should be renamed. No point have "unlimited" if there's no "limited". :)